cpp/src/slice2swift/Gen.cpp

```diff
@@ -526,7 +526,7 @@ Gen::TypesVisitor::visitExceptionStart(const ExceptionPtr& p)
     }
     out << eb;
 
-    if (p->usesClasses(false) && (!base || (base && !base->usesClasses(false))))
+    if (p->usesClasses() && (!base || (base && !base->usesClasses())))
```

Comment:
Seems we can simplify the check

```suggestion
    if (p->usesClasses() && (!base || !base->usesClasses()))
```

Does useClasses already check base types?

Reply:
> Does useClasses already check base types?

Yes, but after staring at this for a little while, I think it's checking something different.
This `if` condition only passes if:
- this exception uses classes
- and it's base _**doesn't**_ use classes (or it doesn't have a base)

Reply:
Feel free to double check me here though.